File: source/code/php/authentication/create_directory_saml.php
$provider = $client->dataStore->instantiate(\Stormpath\Stormpath::SAML_PROVIDER);
$provider->ssoLoginUrl = "http://provider.com/login";
$provider->ssoLogoutUrl = "http://provider.com/logout";
$provider->encodedX509SigningCert = '-----BEGIN CERTIFICATE-----\nMIIDX...tUJ+dp8aJnprxyB\nnQ==\n-----END CERTIFICATE-----';
$provider->requestSignatureAlgorithm = 'RSA-SHA1';

$directory = $client->dataStore->instantiate(\Stormpath\Stormpath::DIRECTORY);
$directory->name = "my-saml-directory";
$directory->description = "A Saml directory";
$directory->provider = $provider;

$tenant = $client->getCurrentTenant();
$directory = $tenant->createDirectory($directory);

File: source/code/python/authentication/create_directory_saml.py
from stormpath.resources import Provider

client.directories.create({
    'name': 'My SAML Directory',
    'description': 'A Directory used for SAML Authorization',
    'provider': {
        'provider_id': Provider.SAML,
        'sso_login_url': 'https://example.com/saml2/sso/login',
        'sso_logout_url': 'https://example.com/saml2/sso/logout',
        'encoded_x509_signing_cert': '-----BEGIN CERTIFICATE-----\n...Certificate goes here...\n-----END CERTIFICATE-----',
        'request_signature_algorithm': Provider.SIGNING_ALGORITHM_RSA_SHA_256,
    },
})
